From 8903bd731e26b578d16323c01dc32d7770596b51 Mon Sep 17 00:00:00 2001
From: Hao Zheng <haoz@marvell.com>
Date: Thu, 20 Jun 2019 16:52:22 -0700
Subject: [PATCH 303/386] octeontx2-af: fix parser error for udp packets with
 no payload

fix npc parser error when parsing udp packets that don't have
any payload

Change-Id: I6be1113cee968fb49b23e2a44dab8e1e2b7d4ae1
Signed-off-by: Hao Zheng <haoz@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11393
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stanislaw Kardach <Stanislaw.Kardach@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
index 16b8f767607f..2f975a3a552e 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/npc_profile.h
@@ -8925,7 +8925,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		2, 8, 10, 2, 0,
+		2, 0, 0, 2, 0,
 		NPC_S_KPU8_UDP, 20, 1,
 		NPC_LID_LC, NPC_LT_LC_IP,
 		0,
@@ -9181,7 +9181,7 @@ static struct npc_kpu_profile_action kpu5_action_entries[] = {
 	},
 	{
 		NPC_ERRLEV_RE, NPC_EC_NOERR,
-		2, 8, 10, 2, 0,
+		2, 0, 0, 2, 0,
 		NPC_S_KPU8_UDP, 40, 1,
 		NPC_LID_LC, NPC_LT_LC_IP6,
 		0,
-- 
2.17.1

